pkg/notify/notifier/pushover/pushover.go

```diff
@@ -0,0 +1,267 @@
+		if len(filteredData.Alerts) == 0 {
+			return nil
+		}
+
+		// split new data along with its Alerts to ensure each message is small enough to fit the Pushover's message length limit
+		messages, _, err := n.template.Split(filteredData, MessageMaxLength, c.Template, "", n.logger)
+		if err != nil {
+			_ = level.Error(n.logger).Log("msg", "PushoverNotifier: split alerts error", "userKey", userKey, "error", err.Error())
+			return err
+		}
+
+		// send messages in parallel
+		g := new(errgroup.Group)
+		for _, message := range messages {
+			msg := message
+			g.Go(func() (err error) {
+				// consume a token from limiter bucket
+				if err := n.limiter.Wait(context.Background()); err != nil {
+					_ = level.Error(n.logger).Log("msg", "PushoverNotifier: limiter error", "userKey", userKey, "error", err.Error())
+					return err
+				}
+
+				// construct pushover message struct as request parameters
+				pReq := &pushoverRequest{
+					Token:   token,
+					UserKey: userKey,
+					Message: msg,
+				}
+				if len(profile.Devices) > 0 {
+					pReq.Device = strings.Join(profile.Devices, ",")
+				}
+				if profile.Title != nil {
+					pReq.Title = *profile.Title
+				}
+				if profile.Sound != nil {
+					pReq.Sound = *profile.Sound
+				}
+
+				// JSON encoding
+				var buf bytes.Buffer
+				if err := utils.JsonEncode(&buf, pReq); err != nil {
+					_ = level.Error(n.logger).Log("msg", "PushoverNotifier: encode message error", "userKey", userKey, "error", err.Error())
+					return err
+				}
+
+				// build a JSON request with context
+				request, err := http.NewRequestWithContext(context.Background(), http.MethodPost, URL, &buf)
+				if err != nil {
+					_ = level.Error(n.logger).Log("msg", "PushoverNotifier: encode http request error", "userKey", userKey, "error", err.Error())
+					return err
+				}
+				request.Header.Set("Content-Type", "application/json")
+
+				// send the request
+				response, err := n.client.Do(request)
+				if err != nil {
+					_ = level.Error(n.logger).Log("msg", "PushoverNotifier: do http error", "userKey", userKey, "error", err.Error())
+					return err
+				}
+
+				defer func() {
+					_, _ = io.Copy(ioutil.Discard, response.Body)
+					_ = response.Body.Close()
+				}()
+
+				// check status code, but not return error if it is not 2xx, since we will do this later
+				if response.StatusCode != http.StatusOK {
+					_ = level.Error(n.logger).Log("msg", "PushoverNotifier: got non-2xx response", "userKey", userKey, "StatusCode", response.StatusCode)
+					body, _ := ioutil.ReadAll(response.Body)
+					return fmt.Errorf("PushoverNotifier: got non-2xx response, StatusCode: %d, response: %s", response.StatusCode, string(body))
+				}
+
+				// report the remaining number of messages for this period
+				_ = level.Info(n.logger).Log("msg", "PushoverNotifier: remaining available message(s) for this period", "userKey", userKey, "remaining", response.Header.Get("X-Limit-App-Remaining"))
```

Comment:
This will generate a lot of logs, change it to `debug`.

Reply:
Removed.
Users (a.k.a. message receivers) do not have receving limit. Each platform of a receiver has a $5 USD one-time purchase subscription which allows this platform to receive messages unlimitedly, otherwise a 30-day free trial is applicable. Each application has a message capacity, 10000 by default.